src/game_interpreter_battle.cpp

```diff
@@ -596,21 +597,143 @@ bool Game_Interpreter_Battle::CommandManiacControlBattle(lcf::rpg::EventCommand
 	return true;
 }
 
-bool Game_Interpreter_Battle::CommandManiacControlAtbGauge(lcf::rpg::EventCommand const&) {
+bool Game_Interpreter_Battle::CommandManiacControlAtbGauge(lcf::rpg::EventCommand const& com) {
 	if (!Player::IsPatchManiac()) {
 		return true;
 	}
 
-	Output::Warning("Maniac Patch: Command ControlAtbGauge not supported");
+	// 0 actor
+	// 1 party member
+	// 2 entire party
+	// 3 troop member
+	// 4 entire troop
+	int targetFlags = com.parameters[0];
+	// 0 constant
+	// 1 variable
+	// 2 variable id
+	int targetReferenceFlags = com.parameters[1];
+	int targetReferenceIdentifier = com.parameters[2];
+	// 0 set
+	// 1 add
+	// 2 sub
+	int operationFlags = com.parameters[3];
+	// 0 value
+	// 1 percentage
+	int operandFlags = com.parameters[4];
+	// 0 constant
+	// 1 variable
+	// 2 variable id
+	int valueReferenceFlags = com.parameters[5];
+	int valueReferenceIdentifier = com.parameters[6];
+
+	auto getVariableOrValue = [](int flags, int identifier) {
```

Comment:
We already have a function ``ValueOrVariable`` that does this. Also has the same signature ``(mode, value)``. :)